hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go

```diff
@@ -159,6 +160,24 @@ func (a *Azure) ReconcileCredentials(ctx context.Context, c client.Client, creat
 		}
 		return nil
 	})
+
+	// Sync CNCC secret
+	name = client.ObjectKey{Namespace: "openshift-cloud-network-config-controller", Name: "cloud-credentials"}
```

Comment:
why is this looking this up in the management cluster?

Reply:
I was trying to follow the idea from https://github.com/openshift/hypershift/pull/1679/files#diff-06207575f3cfe4069b485056885b8807a7ad0f29e07948aa573434531aa9665cL151 where this secret needs to be in the CPO namespace for the CNCC pod to initialize. Maybe this is not the right way to get that secret over to the CPO namespace?

Reply:
for aws we pass each granular perm, for azure we don't have granularity atm all reconciled secret should source from https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go#L147, that's already used for capz and the cloud controller manager

Reply:
basically we just need to generate a secret named cloud-network-config-controller-creds out of the original one, the likes of
```
	cloudCreds := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: controlPlaneNamespace, Name: name.Name}}
	cloudNetworkConfigCreds := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: controlPlaneNamespace, Name: "cloud-network-config-controller-creds"}}
	targets := []*corev1.Secret{cloudNetworkConfigCreds, cloudCreds}
	for _, t := range targets {
		_, err := createOrUpdate(ctx, c, t, func() error {
			if t.Data == nil {
				t.Data = map[string][]byte{}
			}
			for k, v := range source.Data {
				t.Data[k] = v
			}
			return nil
		})
		return err
	}
```

and we can most likely drop the one from resources for the guest cluster.